lnbits/helpers.py

```diff
@@ -166,3 +170,20 @@ def update_query(table_name: str, model: BaseModel, where: str = "WHERE id = ?")
     """
     query = ", ".join([f"{field} = ?" for field in model.dict().keys()])
     return f"UPDATE {table_name} SET {query} {where}"
+
+
+def is_valid_email_address(email: str) -> bool:
+    email_regex = r"\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,7}\b"
+    return re.fullmatch(email_regex, email) is not None
+
+
+def is_valid_username(username: str) -> bool:
+    username_regex = r"\b(?=[a-zA-Z0-9._]{2,20}$)(?!.*[_.]{2})[^_.].*[^_.]\b"
```

Comment:
I am clueless. Where does this come from and why is it so complicated?

Reply:
Adapted from here: https://stackoverflow.com/questions/12018245/regular-expression-to-validate-username

```
- allowed characters: a-zA-Z0-9._
- username is 2-20 characters long
- no __ or _. or ._ or .. inside
- no _ or . at the beginning
- no _ or . at the end
```